tower/src/retry/budget/tps_budget.rs

```diff
@@ -20,25 +20,23 @@ use super::{Bucket, BudgetTrait};
 /// For more info about [`Budget`], please see the [module-level documentation].
 ///
 /// [module-level documentation]: super
-pub struct Budget<B = TpsBucket> {
-    bucket: B,
+pub struct TpsBudget {
+    bucket: TpsBucket,
```

Comment:
I would remove this extra struct and just merge everything into the TpsBucket and rename that TpsBudget. That should simplify some of this code and also make it less confusing since TpsBucket looks like TpsBudget lol